applications/GeoMechanicsApplication/custom_processes/find_neighbour_elements_of_conditions_process.hpp

```diff
@@ -8,138 +8,83 @@
+using SortedToUnsortedNodeIdsHashMap = std::unordered_multimap<std::vector<std::size_t>,
+                                                               std::vector<std::size_t>,
+                                                               KeyHasherRange<std::vector<std::size_t>>,
+                                                               KeyComparorRange<std::vector<std::size_t>>>;
+
 class KRATOS_API(GEO_MECHANICS_APPLICATION) FindNeighbourElementsOfConditionsProcess : public Process
 {
 public:
-    ///@name Type Definitions
-    ///@{
-
-    /// Pointer definition of Process
     KRATOS_CLASS_POINTER_DEFINITION(FindNeighbourElementsOfConditionsProcess);
 
-    /// The definition of the index type
-    using IndexType = std::size_t;
-
-    /// Definition of the node type
-    using NodeType = Node;
-
-    // Definition of the geometry
-    using GeometryType = Geometry<NodeType>;
-
-    ///@}
-    ///@name Life Cycle
-    ///@{
-
-    /// Constructor for FindNeighbourElementsOfConditionsProcess Process
-    /**
-     * @param rModelPart The model part to check.
-     */
     explicit FindNeighbourElementsOfConditionsProcess(ModelPart& rModelPart)
-        : Process(), mrModelPart(rModelPart)
+        : mrModelPart(rModelPart)
     {
     }
 
     FindNeighbourElementsOfConditionsProcess& operator=(const FindNeighbourElementsOfConditionsProcess&) = delete;
     FindNeighbourElementsOfConditionsProcess(const FindNeighbourElementsOfConditionsProcess&) = delete;
     ~FindNeighbourElementsOfConditionsProcess() override = default;
 
-    ///@}
-    ///@name Operators
-    ///@{
-
-    ///@}
-    ///@name Operations
-    ///@{
-
-    /// Finds neighbour elements of conditions
     void Execute() override;
 
-    ///@}
-    ///@name Access
-    ///@{
-
-    ///@}
-    ///@name Inquiry
-    ///@{
-
-    ///@}
-    ///@name Input and output
-    ///@{
-
-    /// Turn back information as a string.
-    std::string Info() const override { return "FindNeighbourElementsOfConditionsProcess"; }
+    [[nodiscard]] std::string Info() const override
+    {
+        return "FindNeighbourElementsOfConditionsProcess";
+    }
 
-    /// Print object's data.
     void PrintData(std::ostream& rOStream) const override { this->PrintInfo(rOStream); }
 
-    ///@}
 private:
-    ///@name Static Member Variables
-    ///@{
-    hashmap::iterator FindFaceReorderingTetrahedra3D10(DenseVector<int> FaceIds, hashmap& FacesMap) const;
-    hashmap::iterator FindFaceReorderingTetrahedra3D4(DenseVector<int> FaceIds, hashmap& FacesMap) const;
-    hashmap::iterator FindFaceReorderingHexahedra3D8(DenseVector<int> FaceIds, hashmap& FacesMap) const;
-    hashmap::iterator FindFaceReorderingHexahedra3D20(DenseVector<int> FaceIds, hashmap& FacesMap) const;
-
-    ///@}
-    ///@name Member Variables
-    ///@{
+    ModelPart&                     mrModelPart;
+    NodeIdToConditionsHashMap      mConditionNodeIdsToCondition;
```

Comment:
I would suggest to make the data member's name reflect that the value associated with each key is a collection of condition pointers:
```suggestion
    NodeIdToConditionsHashMap      mConditionNodeIdsToConditions;
```

Reply:
Yes, done